File: src/main/java/com/trade/securities/adapter/in/web/BondTradingInfoController.java
package com.trade.securities.adapter.in.web;

import com.trade.securities.application.port.in.GetBondTradingInfoUseCase;
import com.trade.securities.domain.BondTradingInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/bond")
@RequiredArgsConstructor
public class BondTradingInfoController {

    private final GetBondTradingInfoUseCase getBondTradingInfoUseCase;

    @GetMapping("/trading-info/{marketType}")
    public List<BondTradingInfo> getBondTradingInfo(
            @PathVariable String marketType,
            @RequestParam String basDd) {
        return getBondTradingInfoUseCase.getBondTradingInfo(marketType, basDd);
    }
}

File: src/main/kotlin/com/trade/securities/adapter/in/web/BondTradingInfoController.kt
package com.trade.securities.adapter.`in`.web

import com.trade.securities.application.port.`in`.GetBondTradingInfoUseCase
import com.trade.securities.domain.BondTradingInfo
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import jakarta.ws.rs.Produces
import jakarta.ws.rs.QueryParam
import jakarta.ws.rs.core.MediaType

@ApplicationScoped
@Path("/api/bond")
@Produces(MediaType.APPLICATION_JSON)
class BondTradingInfoController(
    private val getBondTradingInfoUseCase: GetBondTradingInfoUseCase
) {

    @GET
    @Path("/trading-info/{marketType}")
    fun getBondTradingInfo(
        @PathParam("marketType") marketType: String,
        @QueryParam("basDd") basDd: String
    ): List<BondTradingInfo> {
        return getBondTradingInfoUseCase.getBondTradingInfo(marketType, basDd)
    }
}
